packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -113,9 +118,9 @@ export function makeVirtualReferenceManager(
       // we must ensure addToPossiblyDeadSet only happens once).
       case 'recognizable': {
         syscall.vatstoreSet(key, '0');
-        const refCount = getRefCount(vobjID);
+        const refCount = getRefCount(slot);
```

Comment:
This is probably correct, but the types are now muddled. `getRefCount()` takes `vobjID` and parses out `slot` before touching vatstore, but in this case `setExportStatus` is doing the same thing, so it's kinda being transformed twice, and that smells wrong.

I'm trying to think of better names for the arguments in all these functions, but nothing is coming to mind. Multi-facet virtual objects live or die as a "cohort". It'd be easier to reason about if single-facet virtual objects lived in a cohort of size one. Then vrefs point to specific Representatives (facets), there's a function to extract the "cohort ID" from a vref (possibly a no-op), reference counts are on cohort IDs.

Oh, something which does feel more specifically wrong: why is `getExportStatus` looking at the cohort as a whole? The kernel doesn't know that the different facets are different objects, so it's going to do `dispatch.dropExports` on them individually (and we're going to re-export the facets at different times). That feels like it means we need to track the exported/recognizable/neither status for each vref (facet) separately.

Let's see, that's not the whole story. If one facet of the cohort is kept alive by the kernel, but all the other ones have been dropped by everybody, then we still need to keep the state alive. That facet might have the ability to reach the other facets during actualization, which means the refcounts on the other facets is immaterial: we must be able to recreate them if/when the one exported facet becomes real again.

Ugh, I'm going to have to think about this, maybe we can talk about it tomorrow.

Reply:
As we discussed, it turns out we do need to track export status independently for each facet.  The latest rev now does that.